src/libhoney.js

```diff
@@ -269,14 +272,24 @@ export default class Libhoney extends EventEmitter {
   }
 
   /**
-   * isClassic takes a writeKey and returns true if it is a "classic" writeKey,
-   * namely, that its length is exactly 32 characters.
+   * isClassic takes an API key and returns true if it is a "classic" Configuration API Key or Ingest API Key.
    * @returns {boolean} whether the key is classic
    *
-   * @private
+   * @example
+   *   if(libhoney.isClassic(apiKey)) {
+   *     // special case for classic environments
+   *   }
    */
-  isClassic(key) {
-    return key.length === 32;
+  static isClassic(key) {
+    if (key === null || key === undefined || key.length === 0) {
+      return true;
+    }
+    else if(key.length === 32) {
+      return classicKeyRegex.test(key);
+    } else if(key.length === 64) {
+      return ingestClassicKeyRegex.test(key);
+    }
```

Comment:
I setup + ran some benchmarks locally using [jest-bench](https://github.com/pckhoi/jest-bench). I didn't include the setup in this PR (but would be happy to). The results:

Base (what was here before, just the length check)
```
  isClassic check
    full ingest key string, non classic    0.003 μs ±  0.80 %  (94 runs sampled)
    ingest key id, non classic             0.003 μs ±  1.02 %  (93 runs sampled)
    full ingest key string, classic        0.003 μs ±  0.90 %  (94 runs sampled)
    ingest key id, classic                 0.003 μs ±  1.09 %  (91 runs sampled)
    v2 configuration key                   0.003 μs ±  1.08 %  (94 runs sampled)
    classic key                            0.003 μs ±  0.91 %  (94 runs sampled)
```

With this change:

```
  isClassic check
    full ingest key string, non classic    0.011 μs ±  0.47 %   (98 runs sampled)
    ingest key id, non classic             0.013 μs ±  0.24 %   (99 runs sampled)
    full ingest key string, classic        0.046 μs ±  0.33 %   (95 runs sampled)
    ingest key id, classic                 0.013 μs ±  0.50 %   (99 runs sampled)
    v2 configuration key                   0.004 μs ±  0.24 %   (92 runs sampled)
    classic key                            0.032 μs ±  0.37 %  (100 runs sampled)
```

Reply:
I want to make sure there's no confusion -- what I benchmarked was one implementation of a regular expression that used counting (`{58}`) vs one that didn't depend on counting. Pulling out the counting is what made it fast.

It looks like here you're comparing counting only with counting + regexp, which is going to be slower. That said, sub-microsecond times seem *fine* to me. It would be interesting to see the same benchmark data for asking the regexp to do the counting. 

In any case, for Go the regexp is precompiled, and the Go regexp state machine is [specifically designed to be different](https://swtch.com/~rsc/regexp/regexp1.html) from those in many other languages. The same optimizations might not apply - but maybe they do. It's worth checking.